lambda-functions/database-factory-horizontal-scaling/main.go

```diff
@@ -0,0 +1,331 @@
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to request a new RDS cluster deployment")
+				}
+			}
+		}
+
+	}
+
+	return nil, nil
+}
+
+func requestDeployCluster(vpc string) error {
+	applyBool, err := strconv.ParseBool(os.Getenv("TerraformApply"))
+	if err != nil {
+		return errors.Wrap(err, "failed to parse bool from TerraformApply string")
+	}
+
+	requestBody, err := json.Marshal(map[string]interface{}{
+		"vpcID":                 vpc,
+		"environment":           os.Getenv("Environment"),
+		"stateStore":            os.Getenv("StateStore"),
+		"apply":                 applyBool,
+		"instanceType":          os.Getenv("DBInstanceType"),
+		"backupRetentionPeriod": os.Getenv("BackupRetentionPeriod"),
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal request body")
+	}
+
+	resp, err := http.Post(os.Getenv("DatabaseFactoryEndpoint"), "application/json", bytes.NewBuffer(requestBody))
+	if err != nil {
+		return errors.Wrap(err, "failed to POST request")
+	}
+
+	defer resp.Body.Close()
+	statusOK := resp.StatusCode >= 200 && resp.StatusCode < 300
+	if !statusOK {
+		return errors.Errorf("Non-OK HTTP status:%v", resp.StatusCode)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed tο read response body")
+	}
+	var databaseFactoryRequest DatabaseFactoryRequest
+	err = json.Unmarshal(body, &databaseFactoryRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed tο unmarshal database factory request")
+	}
+	err = sendMattermostNotification(databaseFactoryRequest, "A new RDS Cluster was successfully requested in the Database Factory.")
+	if err != nil {
+		return errors.Wrap(err, "failed tο send Mattermost notification")
+	}
+	log.Infof("The request was posted successfully %s", string(body))
+
+	return nil
+}
+
+func checkVPCScaling(dbClusters *resourcegroupstaggingapi.GetResourcesOutput, vpc string) (bool, error) {
+	var currentClusters int
+	var limitCheck int
+	for _, cluster := range dbClusters.ResourceTagMappingList {
+		var counter *string
+		clusterARN, err := arn.Parse(*cluster.ResourceARN)
+		if err != nil {
+			return false, errors.Wrap(err, "failed tο parse DB cluster ARN")
+		}
+		if !strings.Contains(clusterARN.Resource, os.Getenv("RDSMultitenantDBClusterNamePrefix")) {
+			log.Warnf("Provisioner skipped RDS resource (%s) because name does not have a correct multitenant database prefix (%s)", clusterARN.Resource, os.Getenv("RDSMultitenantDBClusterNamePrefix"))
+			continue
+		}
+		currentClusters++
+		for _, tag := range cluster.Tags {
+			if *tag.Key == "Counter" && tag.Value != nil {
+				counter = tag.Value
+			}
+		}
+		counterInt, err := strconv.Atoi(*counter)
+		if err != nil {
+			return false, errors.Wrap(err, "failed tο change tag counter value into integer")
+		}
+		counterLimit, err := strconv.Atoi(os.Getenv("CounterLimit"))
+		if counterInt >= counterLimit {
+			limitCheck++
+		}
+	}
+	if currentClusters == limitCheck {
```

Comment:
I made some changes to the variable names to make it easier to understand what each does. In this case we actually need this to be equal because we are checking that the clusters that reached the limit of installations are equal to the available clusters.